File: examples/async/kucoin_ws_public.py
"""Subscribe to a KuCoin public WebSocket stream."""

import asyncio

from dcex.ws import kucoin


async def main() -> None:
    """Subscribe to KuCoin spot trades and print one message."""
    async with kucoin.public() as ws:
        await ws.subscribe_trades("BTC-USDT-SPOT")
        print(await ws.recv())


if __name__ == "__main__":
    asyncio.run(main())

File: crates/dcex/examples/kucoin_ws_public.rs
use std::time::Duration;

use dcex::ws::kucoin::KucoinPublicWebSocket;

#[tokio::main]
async fn main() -> dcex::Result<()> {
    let mut ws = KucoinPublicWebSocket::new(Duration::from_secs(10))?;
    ws.connect().await?;
    ws.subscribe_trades("BTC-USDT-SPOT").await?;
    println!("{}", ws.recv().await?);
    ws.close().await?;
    Ok(())
}
